projector-server/src/main/kotlin/org/jetbrains/projector/server/ProjectorServer.kt

```diff
@@ -155,35 +153,29 @@ class ProjectorServer private constructor(
       }
     }
 
-    override fun onWsClose(connection: WebSocket) {
+    builder.onWsClose = { connection ->
       // todo: we need more informative message, add parameters to this method inside the superclass
-      val clientSettings = connection.getAttachment<ClientSettings>()!!
-      val connectionTime = (System.currentTimeMillis() - clientSettings.connectionMillis) / 1000.0
-      logger.info { "${clientSettings.address} disconnected, was connected for ${connectionTime.roundToLong()} s." }
+      connection.getAttachment<ClientSettings>()
+        ?.let { clientSettings ->
+          val connectionTime = (System.currentTimeMillis() - clientSettings.connectionMillis) / 1000.0
+          logger.info { "${clientSettings.address} disconnected, was connected for ${connectionTime.roundToLong()} s." }
+        } ?: logger.info { "Disconnected. This client hasn't clientSettings" }
```

Comment:
I suggest clarifying this message because it can be unclear for a user. Something like "this usually happens when the handshake stage didn't have time to be performed (so it seems the client has been connected for a very short time)". Also, it would be cool to output any piece of info available, for example, the client's address that can be obtained like we do here: https://github.com/JetBrains/projector-server/blob/fb51272f5d18084502a6ff2d958b2d4bf1023b69/projector-server/src/main/kotlin/org/jetbrains/projector/server/ProjectorServer.kt#L166

Reply:
Fixed.